aiogram/contrib/middlewares/i18n.py

```diff
@@ -119,22 +119,24 @@ def lazy_gettext(self, singular, plural=None, n=1, locale=None, enable_cache=Fal
         return LazyProxy(self.gettext, singular, plural, n, locale, enable_cache=enable_cache)
 
     # noinspection PyMethodMayBeStatic,PyUnusedLocal
-    async def get_user_locale(self, action: str, args: Tuple[Any]) -> str:
+    async def get_user_locale(self, action: str, args: Tuple[Any]) -> Optional[str]:
         """
         User locale getter
-        You can override the method if you want to use different way of getting user language.
+        You can override the method if you want to use different way of
+        getting user language.
 
         :param action: event name
         :param args: event arguments
-        :return: locale name
+        :return: locale name or None
         """
-        user: types.User = types.User.get_current()
-        locale: Locale = user.locale
+        user: Optional[types.User] = types.User.get_current()
+        locale: Optional[Locale] = user.locale if user else None
 
         if locale:
             *_, data = args
             language = data['locale'] = locale.language
             return language
+        return None
```

Comment:
When method returns None, await on line 152 fails.
`locale = await self.get_user_locale(action, args)` <- awaiting None occurs here

Reply:
We should check for None in `trigger` before awaiting result

Reply:
Please elaborate. we `await` for a coroutine and not the None. the `get_user_locale(...)` may return a `Awaitable[None | str]`, not `None`.